test/box-luatest/gh_8146_space_and_index_name_in_iproto_requests_test.lua

```diff
@@ -0,0 +1,296 @@
+    end)
+    err_msg = "Space '777' does not exist"
+    t.assert_error_msg_content_equals(err_msg, function()
+        c.space[777]:insert{}
+    end)
+    err_msg = "Space '777' does not exist"
+    t.assert_error_msg_content_equals(err_msg, function()
+        c.space[777].index.i:select{}
+    end)
+    err_msg = "No index 'nonexistent' is defined in space 's'"
+    t.assert_error_msg_content_equals(err_msg, function()
+        c.space[cg.s_id].index.nonexistent:select{}
+    end)
+    err_msg = "No index #777 is defined in space 's'"
+    t.assert_error_msg_content_equals(err_msg, function()
+        c.space[cg.s_id].index[777]:select{}
+    end)
+end
+
+g.after_test('test_net_box_conn_space_and_index_wrapper_tables', function(cg)
+    cg.server:exec(function()
+        box.space.s:drop()
+    end)
+end)
+
+g.before_test('test_space_and_index_name_resolution', function(cg)
+    cg.s_id, cg.s1_id, cg.s2_id = cg.server:exec(function()
+        local s = box.schema.create_space('s')
+        s:create_index('i1')
+        s:create_index('i2', {parts = {2}})
+        s:insert{1, 2}
+        s:insert{2, 1}
+        local s1 = box.schema.create_space('s1')
+        s1:create_index('i')
+        s1:insert{1}
+        local s2 = box.schema.create_space('s2')
+        s2:create_index('i')
+        s2:insert{2}
+        return s.id, s1.id
+    end)
+end)
+
+local function inject_select(c, sid, space_name, iid, idx_name, key)
+    local header = msgpack.encode({
+        [box.iproto.key.REQUEST_TYPE] = box.iproto.type.SELECT,
+        [box.iproto.key.SYNC] = c:_next_sync(),
+    })
+    local body = msgpack.encode({
+        [box.iproto.key.SPACE_ID] = sid,
+        [box.iproto.key.SPACE_NAME] = space_name,
+        [box.iproto.key.INDEX_ID] = iid,
+        [box.iproto.key.INDEX_NAME] = idx_name,
+        [box.iproto.key.LIMIT] = 1,
+        [box.iproto.key.KEY] = setmetatable({key}, {__serialize = 'array'}),
+    })
+    local size = msgpack.encode(#header + #body)
+    local request = size..header..body
+    return c:_inject(request)
+end
+
+local function inject_insert_or_replace(c, request, space_name, index_name)
+        local header = msgpack.encode({
+        [box.iproto.key.REQUEST_TYPE] = box.iproto.type[request],
+        [box.iproto.key.SYNC] = c:_next_sync(),
+    })
+    local body = msgpack.encode({
+        [box.iproto.key.SPACE_NAME] = space_name,
+        [box.iproto.key.INDEX_NAME] = index_name,
+        [box.iproto.key.TUPLE] = {3, 3},
+    })
+    local size = msgpack.encode(#header + #body)
+    local request = size..header..body
+    return c:_inject(request)
+end
+
+-- Checks that space and index name resolution works correctly.
+g.test_space_and_index_name_resolution = function(cg)
+    local c = netbox:connect(cg.server.net_box_uri, {fetch_schema = false})
+    if cg.params.connection == 'stream' then
+        c = c:new_stream()
+    end
+
+    t.assert_equals(inject_select(c, cg.s1_id, 's2'), {{2}})
```

Comment:
AFAICS `_inject` doesn't set stream id. You have to do it manually.

Reply:
AFAIC, stream identifier is only needed for transactions.

Reply:
No, streams can be used without transactions. Anyway, this test doesn't make sense to run with and without streams unless `inject_select` sets `stream_id`.